### ISSUE TITLE
bson library has released v0.2.0

### ISSUE THREAD

**Issue by freiguy1** [mongodb-labs/mongo-rust-driver-prototype#140]
I see the [bson](https://github.com/zonyitoo/bson-rs) lib has released v0.2.0 and I can't use the latest versions of it or Serde without this mongo lib upgrading to the newest bson, I believe. Mongo is currently using bson 0.1.4.

**Comment by saghm** [comment]
@kyeah, you mentioned you would take care of this over [here](https://github.com/zonyitoo/bson-rs/pull/34). Need any help with that?

**Comment by kyeah** [comment]
That would be great, @saghm! I'm traveling right now, so it'll be faster if you could take it on.

**Comment by saghm** [comment]
Sounds good!

**Comment by ollivera** [comment]
Any update on this? I also would like to use the latest version of bson.

**Comment by saghm** [comment]
Sorry about this, I got sidetracked by other things. I'll get working on this right away!

**Comment by vkarpov15** [comment]
Gonna be quite tricky because I'd exactly zero people who have contributed to the current iteration of the driver currently have write access to the repo. Perhaps @acmorrow can help?

**Comment by acmorrow** [comment]
@vkarpov15 Discussion is ongoing, appreciate your patience.

**Comment by dariusc93** [comment]
There is no breaking change for using 0.2 in the mongo driver. I am currently using a fork version of mongo for rust with it using bson 0.2 and everything runs fine

**Comment by freiguy1** [comment]
@dariusc93 I did the same thing. It compile right away when i changed the cargo.toml.

**Comment by vkarpov15** [comment]
@dariusc93 @freiguy1 probably worthwhile to fork for now and merge back later - it'll take a while for me to get write access to this repo again and I'd rather not push to crates from a fork unless it's absolutely necessary.

**Comment by kyeah** [comment]
@vkarpov15 I believe we still need to bump the version and push the changes out to crates.io.

**Comment by vkarpov15** [comment]
@saghm @kyeah I'm getting the following issue when I run `cargo package` with rust 1.8. Any ideas?

```
   Compiling textnonce v0.3.0
/home/val/.cargo/registry/src/github.com-88ac128001ac3a9a/textnonce-0.3.0/build.rs:17:33: 17:46 error: mismatched types:
 expected `&mut syntex::Registry`,
    found `&mut inner::syntex::Registry`
(expected struct `syntex::Registry`,
    found struct `inner::syntex::Registry`) [E0308]
/home/val/.cargo/registry/src/github.com-88ac128001ac3a9a/textnonce-0.3.0/build.rs:17         serde_codegen::register(&mut registry);
                                                                                                                      ^~~~~~~~~~~~~
/home/val/.cargo/registry/src/github.com-88ac128001ac3a9a/textnonce-0.3.0/build.rs:17:33: 17:46 help: run `rustc --explain E0308` to see a detailed explanation
error: aborting due to previous error
failed to verify package tarball

Caused by:
  Could not compile `textnonce`.

To learn more, run the command again with --verbose.
```

Also, do y'all have accounts on crates.io? Would make more sense for you guys to have access to do releases - I don't want to sit around and be a gatekeeper.

**Comment by saghm** [comment]
Ugh, looks like one of the dependencies got broken with the latest Rust update. I'll take a look to see if it's easily fixable.

I do have a crates.io account; it's linked to my github (saghmrossi@gmail.com). I agree, it probably maks sense to add us on as maintainers of the crate.

**Comment by saghm** [comment]
I'm not getting the issue you seem to be having; maybe you have an old cached version of one of the dependencies that's been updated? I'd try deleting `Cargo.lock` and then trying to rebuild from scratch to see if that fixes it.

However, despite all the tests passing locally for me, Travis seems to having an issue with one of the GridFS tests (https://github.com/mongodb-labs/mongo-rust-driver-prototype/issues/144), which we should probably sort out before publishing a new version.

**Comment by vkarpov15** [comment]
@saghm added you as an owner on crates.io, see https://crates.io/crates/mongodb

**Comment by saghm** [comment]
Oh, strange, I'm getting the error you had now. It seems to only happen when I run `cargo package` (not when I just do `cargo build`.

**Comment by kyeah** [comment]
Also on crates via github -- @kyeah (kevinyeah@utexas.edu).

**Comment by saghm** [comment]
It seems to be an issue in the `textnonce` crate that has been fixed but not just pushed to crates yet. (It works fine when I specify using the latest version from Github rather than from crates.io, but crates doesn't allow publishing packages that use dependencies from both itself and github, for good reason).

I've opened an issue with it, so hopefully it will be fixed soon: https://github.com/mikedilger/textnonce/issues/10

**Comment by saghm** [comment]
The issue with `textnonce` was fixed; version 0.1.4 has been published!
